File: GNSDK/wrappers/gnsdk_csharp/src_wrapper/GnLinkContent.cs

namespace GracenoteSDK {

using System;
using System.Runtime.InteropServices;

/**
* GnLinkContent
*/
public class GnLinkContent : GnObject {
  private HandleRef swigCPtr;

  internal GnLinkContent(IntPtr cPtr, bool cMemoryOwn) : base(gnsdk_csharp_marshalPINVOKE.GnLinkContent_SWIGUpcast(cPtr), cMemoryOwn) {
    swigCPtr = new HandleRef(this, cPtr);
  }

  internal static HandleRef getCPtr(GnLinkContent obj) {
    return (obj == null) ? new HandleRef(null, IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnLinkContent() {
    Dispose();
  }

  public override void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnLinkContent(swigCPtr);
        }
        swigCPtr = new HandleRef(null, IntPtr.Zero);
      }
      GC.SuppressFinalize(this);
      base.Dispose();
    }
  }

      public byte[] DataBuffer
      {
        get
        {
          byte[] dataBytes = new byte[DataSize];
          Marshal.Copy((IntPtr)ContentData, dataBytes, 0, (int)DataSize);
          return dataBytes;
        }
      }
    
  public GnLinkContent(SWIGTYPE_p_unsigned_char contentData, uint dataSize, GnLinkContentType contentType, GnLinkDataType dataType) : this(gnsdk_csharp_marshalPINVOKE.new_GnLinkContent(SWIGTYPE_p_unsigned_char.getCPtr(contentData), dataSize, (int)contentType, (int)dataType), true) {
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

/**
* Retrieves content data buffer size
*/
  public uint DataSize {
    get {
      uint ret = gnsdk_csharp_marshalPINVOKE.GnLinkContent_DataSize_get(swigCPtr);
      return ret;
    } 
  }

/**
* Retrieves content data type
*/
  public GnLinkDataType DataType {
    get {
      GnLinkDataType ret = (GnLinkDataType)gnsdk_csharp_marshalPINVOKE.GnLinkContent_DataType_get(swigCPtr);
      return ret;
    } 
  }

  public IntPtr ContentData {get{IntPtr res = gnsdk_csharp_marshalPINVOKE.GnLinkContent_ContentData_get(swigCPtr); ; return res;}
  }

}

}

File: GNSDK/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnLinkContent.java

package com.gracenote.gnsdk;

/** 
* {@link GnLinkContent} 
*/ 
 
public class GnLinkContent extends GnObject {
  private long swigCPtr;

  protected GnLinkContent(long cPtr, boolean cMemoryOwn) {
    super(gnsdk_javaJNI.GnLinkContent_SWIGUpcast(cPtr), cMemoryOwn);
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnLinkContent obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnLinkContent(swigCPtr);
      }
      swigCPtr = 0;
    }
    super.delete();
  }

  public GnLinkContent(byte[] contentData, long dataSize, GnLinkContentType contentType, GnLinkDataType dataType) throws com.gracenote.gnsdk.GnException {
    this(gnsdk_javaJNI.new_GnLinkContent(contentData, dataSize, contentType.swigValue(), dataType.swigValue()), true);
  }

/** 
* Retrieves content data buffer size 
*/ 
 
  public long dataSize() {
    return gnsdk_javaJNI.GnLinkContent_dataSize(swigCPtr, this);
  }

/** 
* Retrieves content data type 
*/ 
 
  public GnLinkDataType dataType() {
    return GnLinkDataType.swigToEnum(gnsdk_javaJNI.GnLinkContent_dataType(swigCPtr, this));
  }

  public void dataBuffer(byte[] pre_allocated_byte_buffer) {
    gnsdk_javaJNI.GnLinkContent_dataBuffer(swigCPtr, this, pre_allocated_byte_buffer);
  }

}
